packages/frontend-core/src/components/CoreFilterBuilder.svelte

```diff
@@ -0,0 +1,510 @@
+<script>
+  import {
+    Body,
+    Button,
+    Icon,
+    Layout,
+    Select,
+    Helpers,
+    ActionButton,
+  } from "@budibase/bbui"
+  import {
+    FieldType,
+    FilterGroupLogicalOperator,
+    EmptyFilterOption,
+  } from "@budibase/types"
+  import { QueryUtils, Constants } from "@budibase/frontend-core"
+  import { getContext, createEventDispatcher } from "svelte"
+  import FilterField from "./FilterField.svelte"
+
+  const dispatch = createEventDispatcher()
+  const {
+    OperatorOptions,
+    DEFAULT_BB_DATASOURCE_ID,
+    FilterOperator,
+    OnEmptyFilter,
+    FilterValueType,
+  } = Constants
+
+  export let schemaFields
+  export let filters
+  export let tables = []
+  export let datasource
+  export let behaviourFilters = false
+  export let allowBindings = false
+
+  // Review
+  export let bindings
+  export let panel
+  export let toReadable
+  export let toRuntime
+
+  $: editableFilters = filters ? Helpers.cloneDeep(filters) : null
+
+  $: {
+    if (
+      tables.find(
+        table =>
+          table._id === datasource?.tableId &&
+          table.sourceId === DEFAULT_BB_DATASOURCE_ID
+      ) &&
+      !schemaFields.some(field => field.name === "_id")
+    ) {
+      schemaFields = [...schemaFields, { name: "_id", type: "string" }]
+    }
+  }
+
+  const filterOperatorOptions = Object.values(FilterOperator).map(entry => {
+    return { value: entry, label: Helpers.capitalise(entry) }
+  })
+
+  const onEmptyOptions = Object.values(OnEmptyFilter).map(entry => {
+    return { value: entry, label: Helpers.capitalise(entry) }
```

Comment:
Could we update the labels to maybe say `All rows` and `No rows`? Just think that reads a little better.